File: crossecore-java/src/java_/ModelBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ModelBase 
extends org.eclipse.emf.ecore.impl.MinimalEObjectImpl.Container implements Model
{
	protected static final java.lang.String NAME_EDEFAULT = "";
	private java.lang.String _name = NAME_EDEFAULT;
	public java.lang.String getName()
	{
		return _name;
	}
	public void setName(java.lang.String value){
		
		java.lang.String oldValue = _name;
		_name = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.MODEL__NAME, oldValue, value));
		
	}
	
	
	
	private com.crossecore.ocl.Set<Package> _ownedElements;
	
	public com.crossecore.ocl.Set<Package> getOwnedElements()
	{
		if(_ownedElements==null){
			_ownedElements = new com.crossecore.ocl.Set<Package>(Package.class, this, Java_PackageImpl.MODEL__OWNED_ELEMENTS, Java_PackageImpl.PACKAGE__MODEL);
		}
		return _ownedElements;
	
	}
	private com.crossecore.ocl.Set<Type> _orphanTypes;
	
	public com.crossecore.ocl.Set<Type> getOrphanTypes()
	{
		if(_orphanTypes==null){
			_orphanTypes = new com.crossecore.ocl.Set<Type>(Type.class, this, Java_PackageImpl.MODEL__ORPHAN_TYPES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MODEL__ORPHAN_TYPES);
		}
		return _orphanTypes;
	
	}
	private com.crossecore.ocl.Set<UnresolvedItem> _unresolvedItems;
	
	public com.crossecore.ocl.Set<UnresolvedItem> getUnresolvedItems()
	{
		if(_unresolvedItems==null){
			_unresolvedItems = new com.crossecore.ocl.Set<UnresolvedItem>(UnresolvedItem.class, this, Java_PackageImpl.MODEL__UNRESOLVED_ITEMS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MODEL__UNRESOLVED_ITEMS);
		}
		return _unresolvedItems;
	
	}
	private com.crossecore.ocl.Set<CompilationUnit> _compilationUnits;
	
	public com.crossecore.ocl.Set<CompilationUnit> getCompilationUnits()
	{
		if(_compilationUnits==null){
			_compilationUnits = new com.crossecore.ocl.Set<CompilationUnit>(CompilationUnit.class, this, Java_PackageImpl.MODEL__COMPILATION_UNITS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MODEL__COMPILATION_UNITS);
		}
		return _compilationUnits;
	
	}
	private com.crossecore.ocl.Set<ClassFile> _classFiles;
	
	public com.crossecore.ocl.Set<ClassFile> getClassFiles()
	{
		if(_classFiles==null){
			_classFiles = new com.crossecore.ocl.Set<ClassFile>(ClassFile.class, this, Java_PackageImpl.MODEL__CLASS_FILES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MODEL__CLASS_FILES);
		}
		return _classFiles;
	
	}
	private com.crossecore.ocl.Set<Archive> _archives;
	
	public com.crossecore.ocl.Set<Archive> getArchives()
	{
		if(_archives==null){
			_archives = new com.crossecore.ocl.Set<Archive>(Archive.class, this, Java_PackageImpl.MODEL__ARCHIVES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MODEL__ARCHIVES);
		}
		return _archives;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.MODEL;
	}
	
	@Override
	public NotificationChain eInverseAdd(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.MODEL__OWNED_ELEMENTS:
				return getOwnedElements().basicAdd((Package)otherEnd, msgs);
		}
		return super.eInverseAdd(otherEnd, featureID, msgs);
	}
	
	@Override
	public NotificationChain eInverseRemove(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.MODEL__OWNED_ELEMENTS:
				return getOwnedElements().basicRemove((Package)otherEnd, msgs);
		}
		return super.eInverseRemove(otherEnd, featureID, msgs);
	}
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.MODEL__NAME:
				return getName();
			case Java_PackageImpl.MODEL__OWNED_ELEMENTS:
				return getOwnedElements();
			case Java_PackageImpl.MODEL__ORPHAN_TYPES:
				return getOrphanTypes();
			case Java_PackageImpl.MODEL__UNRESOLVED_ITEMS:
				return getUnresolvedItems();
			case Java_PackageImpl.MODEL__COMPILATION_UNITS:
				return getCompilationUnits();
			case Java_PackageImpl.MODEL__CLASS_FILES:
				return getClassFiles();
			case Java_PackageImpl.MODEL__ARCHIVES:
				return getArchives();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.MODEL__NAME:
				setName((java.lang.String) newValue);
				return;
			case Java_PackageImpl.MODEL__OWNED_ELEMENTS:
				getOwnedElements().clear();
				getOwnedElements().addAll((java.util.Collection<? extends Package>) newValue);
				return;
			case Java_PackageImpl.MODEL__ORPHAN_TYPES:
				getOrphanTypes().clear();
				getOrphanTypes().addAll((java.util.Collection<? extends Type>) newValue);
				return;
			case Java_PackageImpl.MODEL__UNRESOLVED_ITEMS:
				getUnresolvedItems().clear();
				getUnresolvedItems().addAll((java.util.Collection<? extends UnresolvedItem>) newValue);
				return;
			case Java_PackageImpl.MODEL__COMPILATION_UNITS:
				getCompilationUnits().clear();
				getCompilationUnits().addAll((java.util.Collection<? extends CompilationUnit>) newValue);
				return;
			case Java_PackageImpl.MODEL__CLASS_FILES:
				getClassFiles().clear();
				getClassFiles().addAll((java.util.Collection<? extends ClassFile>) newValue);
				return;
			case Java_PackageImpl.MODEL__ARCHIVES:
				getArchives().clear();
				getArchives().addAll((java.util.Collection<? extends Archive>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.MODEL__NAME:
				return NAME_EDEFAULT == null ? _name != null : !NAME_EDEFAULT.equals(_name);
			case Java_PackageImpl.MODEL__OWNED_ELEMENTS:
				return _ownedElements != null && !_ownedElements.isEmpty();
			case Java_PackageImpl.MODEL__ORPHAN_TYPES:
				return _orphanTypes != null && !_orphanTypes.isEmpty();
			case Java_PackageImpl.MODEL__UNRESOLVED_ITEMS:
				return _unresolvedItems != null && !_unresolvedItems.isEmpty();
			case Java_PackageImpl.MODEL__COMPILATION_UNITS:
				return _compilationUnits != null && !_compilationUnits.isEmpty();
			case Java_PackageImpl.MODEL__CLASS_FILES:
				return _classFiles != null && !_classFiles.isEmpty();
			case Java_PackageImpl.MODEL__ARCHIVES:
				return _archives != null && !_archives.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ModelBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {AbstractCollection} from "ecore/AbstractCollection";
import {Bag} from "ecore/Bag";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {CompilationUnit} from "./CompilationUnit";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {Type} from "./Type";
import {ClassFile} from "./ClassFile";
import {Archive} from "./Archive";
import {Model} from "./Model";
import {UnresolvedItem} from "./UnresolvedItem";
import {Package} from "./Package";
		
			export class ModelBase
			extends BasicEObjectImpl
			implements Model
			{
				private _name:string = '';
				get name():string{
					return this._name;
				}
				set name(value:string){
					this._name = value; 
				}
				private _ownedElements:Set<Package> = null;
				
				get ownedElements():Set<Package>{
					if(this._ownedElements===null){
						this._ownedElements = new Set<Package>(this, Java_PackageLiterals.MODEL_OWNEDELEMENTS, Java_PackageLiterals.PACKAGE_MODEL);
							
					}
					return this._ownedElements;
					
				}
				
				
				private _orphanTypes:Set<Type> = null;
				
				get orphanTypes():Set<Type>{
					if(this._orphanTypes===null){
						this._orphanTypes = new Set<Type>(this, Java_PackageLiterals.MODEL_ORPHANTYPES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MODEL_ORPHANTYPES);
							
					}
					return this._orphanTypes;
					
				}
				
				
				private _unresolvedItems:Set<UnresolvedItem> = null;
				
				get unresolvedItems():Set<UnresolvedItem>{
					if(this._unresolvedItems===null){
						this._unresolvedItems = new Set<UnresolvedItem>(this, Java_PackageLiterals.MODEL_UNRESOLVEDITEMS, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MODEL_UNRESOLVEDITEMS);
							
					}
					return this._unresolvedItems;
					
				}
				
				
				private _compilationUnits:Set<CompilationUnit> = null;
				
				get compilationUnits():Set<CompilationUnit>{
					if(this._compilationUnits===null){
						this._compilationUnits = new Set<CompilationUnit>(this, Java_PackageLiterals.MODEL_COMPILATIONUNITS, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MODEL_COMPILATIONUNITS);
							
					}
					return this._compilationUnits;
					
				}
				
				
				private _classFiles:Set<ClassFile> = null;
				
				get classFiles():Set<ClassFile>{
					if(this._classFiles===null){
						this._classFiles = new Set<ClassFile>(this, Java_PackageLiterals.MODEL_CLASSFILES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MODEL_CLASSFILES);
							
					}
					return this._classFiles;
					
				}
				
				
				private _archives:Set<Archive> = null;
				
				get archives():Set<Archive>{
					if(this._archives===null){
						this._archives = new Set<Archive>(this, Java_PackageLiterals.MODEL_ARCHIVES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MODEL_ARCHIVES);
							
					}
					return this._archives;
					
				}
				
				

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ModelBase.eStaticClass;
				}
			
				public eInverseAdd(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain): NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.MODEL_OWNEDELEMENTS:
							return this.ownedElements.basicAdd(otherEnd as Package, msgs);
					}
					//return this.eInverseAddFromBasicEObjectImpl(otherEnd, featureID, msgs);
					return super.eInverseAdd(otherEnd, featureID, msgs);
				}
				//public eInverseAddFromModel = this.eInverseAdd;
				
				public eInverseRemove(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain):NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.MODEL_OWNEDELEMENTS:
							return this.ownedElements.basicRemove(otherEnd as Package, msgs);
					}
					//return this.eInverseRemoveFromBasicEObjectImpl(otherEnd, featureID, msgs);
					return super.eInverseRemove(otherEnd, featureID, msgs);
				}
				
				//public eInverseRemoveFromModel = this.eInverseRemove;
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.MODEL_NAME:
							return this.name;
						case Java_PackageLiterals.MODEL_OWNEDELEMENTS:
							return this.ownedElements;
						case Java_PackageLiterals.MODEL_ORPHANTYPES:
							return this.orphanTypes;
						case Java_PackageLiterals.MODEL_UNRESOLVEDITEMS:
							return this.unresolvedItems;
						case Java_PackageLiterals.MODEL_COMPILATIONUNITS:
							return this.compilationUnits;
						case Java_PackageLiterals.MODEL_CLASSFILES:
							return this.classFiles;
						case Java_PackageLiterals.MODEL_ARCHIVES:
							return this.archives;
					}
					//return this.eGetFromBasicEObjectImpl(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.MODEL_NAME:
							this.name = <string> newValue;
							return;
						case Java_PackageLiterals.MODEL_OWNEDELEMENTS:
							this.ownedElements.clear();
							this.ownedElements.addAll(newValue);
							return;
						case Java_PackageLiterals.MODEL_ORPHANTYPES:
							this.orphanTypes.clear();
							this.orphanTypes.addAll(newValue);
							return;
						case Java_PackageLiterals.MODEL_UNRESOLVEDITEMS:
							this.unresolvedItems.clear();
							this.unresolvedItems.addAll(newValue);
							return;
						case Java_PackageLiterals.MODEL_COMPILATIONUNITS:
							this.compilationUnits.clear();
							this.compilationUnits.addAll(newValue);
							return;
						case Java_PackageLiterals.MODEL_CLASSFILES:
							this.classFiles.clear();
							this.classFiles.addAll(newValue);
							return;
						case Java_PackageLiterals.MODEL_ARCHIVES:
							this.archives.clear();
							this.archives.addAll(newValue);
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
